src/Analyser/NodeScopeResolver.php

```diff
@@ -6204,25 +6219,43 @@ private function produceArrayDimFetchAssignValueToWrite(array $dimFetchStack, ar
 				continue;
 			}
 
-			if ($scope->hasExpressionType($arrayDimFetch)->yes()) { // keep list for $list[$index] assignments
+			if (!$arrayDimFetch->dim instanceof BinaryOp\Plus) {
```

Comment:
@ondrejmirtes why is the issue related to Plus? there is no `+` contained in the [related code example](https://github.com/staabm/phpstan-src/blob/84b61d41298b069880779a6508cccca689515ed7/tests/PHPStan/Rules/Comparison/data/pr-4375.php)?

do we also need to cater for

Reply:
ohh.. the Plus is related to the list-inference logic below.

so it was a bug lingering for longer already :)